File: core/src/net/brenig/pixelescape/render/background/DrawableBackgroundLayer.java
package net.brenig.pixelescape.render.background;

import com.badlogic.gdx.scenes.scene2d.utils.Drawable;
import net.brenig.pixelescape.render.WorldRenderer;

/**
 * Responsible for drawing a layer of the game's background
 */
public class DrawableBackgroundLayer extends AbstractBackgroundLayer {

	private int width;
	private float scrollSpeed;
	private Drawable drawable;

	public DrawableBackgroundLayer(float scrollSpeed, Drawable drawable) {
		this.scrollSpeed = scrollSpeed;
		this.drawable = drawable;
		width = (int) drawable.getMinWidth();
	}

	public DrawableBackgroundLayer(int width, float scrollSpeed, Drawable drawable) {
		this.width = width;
		this.scrollSpeed = scrollSpeed;
		this.drawable = drawable;
	}

	@Override
	public void draw(WorldRenderer renderer) {
		renderer.renderDrawable(drawable, getBackgroundWorldStart(renderer, scrollSpeed, width), 0, renderer.getWorld().getWorldWidth(), renderer.getWorld().getWorldHeight());
	}
}

File: core/src/net/brenig/pixelescape/render/background/DrawableBackgroundLayer.kt
package net.brenig.pixelescape.render.background

import com.badlogic.gdx.scenes.scene2d.utils.Drawable
import net.brenig.pixelescape.render.WorldRenderer

/**
 * Responsible for drawing a layer of the game's background
 */
class DrawableBackgroundLayer : AbstractBackgroundLayer {

    private var width: Int = 0
    private var scrollSpeed: Float = 0.toFloat()
    private var drawable: Drawable? = null

    constructor(scrollSpeed: Float, drawable: Drawable) {
        this.scrollSpeed = scrollSpeed
        this.drawable = drawable
        width = drawable.minWidth.toInt()
    }

    constructor(width: Int, scrollSpeed: Float, drawable: Drawable) {
        this.width = width
        this.scrollSpeed = scrollSpeed
        this.drawable = drawable
    }

    override fun draw(renderer: WorldRenderer) {
        renderer.renderDrawable(drawable!!, getBackgroundWorldStart(renderer, scrollSpeed, width), 0f, renderer.world.worldWidth.toFloat(), renderer.world.worldHeight.toFloat())
    }
}
